pkg/node/publish_unpublish.go

```diff
@@ -78,6 +84,17 @@ func (n *NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublish
 	}
 	vol.Status.ContainerPath = containerPath
 
+	volumeContext, volumeLabels := req.GetVolumeContext(), vol.ObjectMeta.GetLabels()
+	if volumeLabels == nil {
+		volumeLabels = make(map[string]string)
+	}
+	for k, v := range volumeContext {
+		if strings.Contains(k, directCSIVolumeContextPrefix) {
```

Comment:
@wlan0  Will there be any security vulnerabilities if we add all the keys under this particular domain `csi.storage.k8s.io`? Or we can selectively add only `pod.name` and `pod.namespace` for now?

Reply:
The prefix we want is `direct.csi.min.io`. Not `csi.storage.k8s.io`

We shouldn't add `pod.name` or `pod.namespace`. We should only add the labels that comes in from the podSpec. Adding podName and podNamespace could be a security vulnerability. However, just adding the info under `direct.csi.min.io/*` labels is not.